File: 119-pascals-triangle-ii/119-pascals-triangle-ii.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
  const ans = new Array(rowIndex + 1).fill(1)
  let i = 1
  let f = rowIndex
  for (; i < ~~((rowIndex+1)/2); i++, f--) {
    const v = ans[i-1] * f / i
    ans[i] = v
    ans[rowIndex-i] = v
  }
  if (rowIndex > 0 && rowIndex % 2 === 0) ans[i] = ans[i-1] * f / i
  return ans
};

File: 119-pascals-triangle-ii/119-pascals-triangle-ii.go
func getRow(rowIndex int) []int {  
  ans := make([]int, rowIndex + 1)
  ans[0] = 1
  ans[rowIndex] = 1
  for i := 1; i <= rowIndex / 2; i++ {
    ans[i] = ans[i-1] * (rowIndex + 1 - i) / i
    ans[rowIndex - i] = ans[i]
  }
  return ans
}